unstructured/documents/elements.py

```diff
@@ -193,22 +192,23 @@ class ElementMetadata:
     # Detection Model Class Probabilities from Unstructured-Inference Hi-Res
     detection_class_prob: Optional[float] = None
 
-    if UNSTRUCTURED_INCLUDE_DEBUG_METADATA:
-        detection_origin: Optional[str] = None
-
-    def __setattr__(self, key, value):
-        if not UNSTRUCTURED_INCLUDE_DEBUG_METADATA and key == "detection_origin":
-            return
-        else:
-            super().__setattr__(key, value)
+    # -- The detection mechanism responsible for this element, for debugging purposes. Should
+    # -- only be populated when UNSTRUCTURED_INCLUDE_DEBUG_METADATA flag is True. Note
```

Comment:
er. but `UNSTRUCTURED_INCLUDE_DEBUG_METADATA` references are gone?

Reply:
This is in the definition of `ElementMetadata`. The initial implementation made the _definition_ of the `detection_origin` field on `ElementMetadata` conditional based on that flag. That means there are two distinct definitions of `ElementMetadata` and every part of the code that uses it (many of course) needs to become aware that there are two definitions and add code to figure out which one is in effect when they use it.

Then this `.__setattr__()` override was needed to prevent raising an exception when the field was not defined.

Benjamin and I discussed and agreed that there's no problem with having the `detection_origin` field defined on `ElementMetadata` just like any other optional metadata field and then it can be considered or not based on the state of that flag. That limits the coupling to those parts of the code that care about that, and allowed removal of some of the broad special-case, effect-at-a-distance, general-complexifying consequences of a conditional definition.

Reply:
@cragwolfe yep, put this back in.